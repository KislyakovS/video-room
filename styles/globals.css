*,
*::after,
*::before {
    box-sizing: border-box;
}

html, body {
    width: 100vw;
    min-height: 100vh;
    margin: 0;
    padding: 0;

    font-family: var(--font-family);
}

:root {
    --container-width: 1240px;
    --container-gap: 15px;
    --header-height: 80px;
    --main-height: calc(100vh - 80px - 20px);

    --active-color: #fb8660;
    --error-color: #f7716e;

    --font-family: 'Nunito', sans-serif;

    --border: 1px solid #e3e3e3;
    --border-radius: 14px;

    --transition-time: .2s;
    --transition-timing: ease-in;
}

.room-main {
    display: flex;
    gap: 20px;
    height: var(--main-height);
}

.room-main > section {
    height: inherit;
}

.room-main > section:first-child {
    width: 70%;
}

.room-main > section:last-child {
    width: 30%;
}

.swiper {
    width: 100%;
    padding-bottom: 20px !important;
}

.swiper .swiper-pagination {
    bottom: 0 !important;
}

.swiper .swiper-pagination-bullet-active {
    background-color: var(--active-color);
}