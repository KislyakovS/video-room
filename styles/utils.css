.w-100 {
    width: 100%;
}

.mb-20 {
    margin-bottom: 20px;
}

.center {
    position: absolute;
    top: 25%;
    left: 50%;

    width: 440px;

    transform: translateX(-50%);
}

.visually-hidden {
    position: absolute;
    width: 1px;
    height: 1px;
    padding: 0;
    margin: -1px;
    overflow: hidden;
    clip: rect(0, 0, 0, 0);
    white-space: nowrap;
    border: 0;
  }