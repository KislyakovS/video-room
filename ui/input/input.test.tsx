import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Input from './input';
import styles from './input.module.css';

describe('<Input>', () => {
    it('snapshot', () => {
        const { container } = render(<Input placeholder='input field' />);

        expect(container).toMatchSnapshot();
    });

    it('snapshot error message', () => {
        const { container } = render(<Input error='Field is not valid' />);

        expect(container).toMatchSnapshot();
    });

    it('snapshot right side element', () => {
        const { container } = render(<Input
            endAdornment={<div>Right Side Element</div>}
        />);

        expect(container).toMatchSnapshot();
    });

    it('should render with the input and fullWidth classes', () => {
        const { getByRole } = render(<Input fullWidth />);
        const input = getByRole('textbox');

        const hasInputClass = input.classList.contains(styles.input);
        const hasFullWidthClass = input.classList.contains(styles.fullWidth);

        expect(hasInputClass).toBeTruthy();
        expect(hasFullWidthClass).toBeTruthy();
    });

    it('should enter a value in input', async () => {
        const INPUT_VALUE = 'Lorem Ipsum is simply dummy text of the printing ';
        const user = userEvent.setup();

        const { getByRole } = render(<Input />);
        const input = getByRole('textbox') as HTMLInputElement;

        await user.type(input, INPUT_VALUE);

        expect(input.value).toBe(INPUT_VALUE);
    })
})