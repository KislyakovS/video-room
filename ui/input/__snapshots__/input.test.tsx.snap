// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <input
      class="input"
      placeholder="input field"
    />
  </div>
</div>
`;

exports[`<Input> snapshot error message 1`] = `
<div>
  <div
    class="wrapper"
  >
    <input
      class="input error"
    />
    <span
      class="errorMessage"
    >
      Field is not valid
    </span>
  </div>
</div>
`;

exports[`<Input> snapshot right side element 1`] = `
<div>
  <div
    class="wrapper"
  >
    <input
      class="input indentRight"
    />
    <div
      class="endAdornment"
    >
      <div>
        Right Side Element
      </div>
    </div>
  </div>
</div>
`;
